packages/sso/src/oidc/discovery.ts

```diff
@@ -279,42 +277,147 @@ export function validateDiscoveryDocument(
 /**
  * Normalize URLs in the discovery document.
  *
- * Phase 1: This is a stub that returns the document unchanged.
- * Future phases will:
- * - Resolve relative URLs against issuer base
- * - Normalize URL formats
- * - Apply security transformations
- *
- * TODO(Phase 2): Implement normalization of relative URLs based on issuer/discovery endpoint.
- * For now, this is an identity function so we can wire discovery into flows incrementally.
- *
  * @param doc - The discovery document
- * @param _issuerBase - The base issuer URL (unused in Phase 1)
+ * @param issuer - The base issuer URL (unused in Phase 1)
+ * @param isTrustedOrigin - Origin verification tester function
  * @returns The normalized discovery document
  */
 export function normalizeDiscoveryUrls(
 	doc: OIDCDiscoveryDocument,
-	_issuerBase: string,
+	issuer: string,
+	isTrustedOrigin: DiscoverOIDCConfigParams["isTrustedOrigin"],
 ): OIDCDiscoveryDocument {
-	// TODO(Phase 2): Implement URL normalization
+	doc.token_endpoint = normalizeAndValidateUrl(
+		"token_endpoint",
+		doc.token_endpoint,
+		issuer,
+		isTrustedOrigin,
+	);
+	doc.authorization_endpoint = normalizeAndValidateUrl(
+		"authorization_endpoint",
+		doc.authorization_endpoint,
+		issuer,
+		isTrustedOrigin,
+	);
+
+	doc.jwks_uri = normalizeAndValidateUrl(
+		"jwks_uri",
+		doc.jwks_uri,
+		issuer,
+		isTrustedOrigin,
+	);
+
+	if (doc.userinfo_endpoint) {
+		doc.userinfo_endpoint = normalizeAndValidateUrl(
+			"userinfo_endpoint",
+			doc.userinfo_endpoint,
+			issuer,
+			isTrustedOrigin,
+		);
+	}
+
+	if (doc.revocation_endpoint) {
+		doc.revocation_endpoint = normalizeAndValidateUrl(
+			"revocation_endpoint",
+			doc.revocation_endpoint,
+			issuer,
+			isTrustedOrigin,
+		);
+	}
+
 	return doc;
 }
 
+/**
+ * Normalizes and validates a single URL endpoint
```

Comment:
Can we fix the JSON Doc to match the params in the function? Missing isTrustedOrigin and what we return

Reply:
Done.